### ISSUE TITLE
Extracting span attributes before span creation and passing them to `start_span`/`start_as_current_span`

### ISSUE THREAD

**Issue by srikanthccv** [open-telemetry/opentelemetry-python-contrib#430]
Need for this: The ability to make sampling decision based on the attributes following semantic conventions.

I would like to customise the sampler and make my sampling decision based on incoming request metadata for web api. The instrumentations now create span with it's name and kind then conditionally populate tags if span is being recorded. If a sampler doesn't receive any metadata about the incoming request/DB call/other then I am being limited to make my decision based on tiny information available which isn't really helpful.

**Comment by lzchen** [comment]
@lonewolf3739 
You are referring to the scenario that the incoming request is from some other out-of-process thing right? So the incoming span could be non-recording (not sampled), but you may want to sample?

**Comment by lonewolf3739** [comment]
Yes, my use-case is out-of-process.

**Comment by lzchen** [comment]
@lonewolf3739 
Perhaps the `TraceState` field can be used for that purpose?

**Comment by lonewolf3739** [comment]
I don't find tracestate to be useful here because It is mainly used/targeted for vendor specific trace information. My simple use-case is (Collect 100% traces for sensitive endpoints based on http metadata + Probabilistic sampling for everything else).

**Comment by lzchen** [comment]
@lonewolf3739 
Right. I think differing sampling decisions across different processes have been a topic of discussion and problem that we have not been able to solve yet. The `is_recording` optimizing of not add any span attributes when the span is not sampled only makes sense within the confines of the current process. I would bring this up in the SIG meeting.

**Comment by srikanthccv** [comment]
Closing in favour of #936

**Comment by lzchen** [comment]
Re-opening as #936 only covers http instrumentations.
